### PR TITLE
Simplify colonoscopy on Windows

### PR DESCRIPTION
**Pull request by PaulWessel**
The solution implemented in #6658 was not correct and too obtuse.  This PR breaks down the checks to dispense with the easy checks up front and save the more involved one for when it is needed.
Hopefully fixes the issue - let the CI begin.

Note: I have only compiled this in Windows in my head...

Also, the second reference to got_slash is not needed (since we only get past the || if got_slash is true anyway) but I left it there for clarity.

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -4,7 +4,7 @@
 name: Tests
 
 on:
-  # pull_request:
+  pull_request:
```
Comment (seisman) [inline .github/workflows/tests.yml:7]: I've enabled the tests in this PR temporarily. Remember to revert this change before merging.
```suggestion
  # pull_request:
```

Comment (PaulWessel) [reply .github/workflows/tests.yml:7]: Looks like those failures are gone.  The other failures are unrelated to this PR.

**Comment by PaulWessel** [comment]
This PR fixes the failures reported on WIndows.

**Comment by seisman** [comment]
> This PR fixes the failures reported on WIndows.

https://github.com/GenericMappingTools/gmt/runs/6271377114?check_suite_focus=true

I think the tests still fail on Windows.

**Comment by PaulWessel** [comment]
How very odd - I just clicked on the link above and found just 3-4 other errors...

**Comment by PaulWessel** [comment]
So before I added your "turn off the full test" commit I clicked on the failing Windows link and found none of the previous errors.  That is why I added in your suggestion and of course that rebuilt the limited tests wich passed.  How is this possible?

**Comment by seisman** [comment]
> How very odd - I just clicked on the link above and found just 3-4 other errors...

I think what you saw are the tests on macOS.

**Comment by PaulWessel** [comment]
OK, now would be a good time for you to run one of these in the debug and tell me why it fails since it seems to me it should not...

**Comment by PaulWessel** [comment]
Pretty sure I clicked the Windows failing link but hard to prove now.  Let's see if @joa-quim can give some insights here first before we reset to full testing.

**Comment by PaulWessel** [comment]
@joa-quim, put your rubber gloves on and get in that colon!

**Comment by joa-quim** [comment]
``got_slash`` was failing. Make it
``got_slash = (strchr ("/\\", text[j+1]) != NULL);	/* True if next letter is a slash */``

**Comment by PaulWessel** [comment]
OK, updated, please run tests

**Comment by joa-quim** [comment]
Tests pass now.

**Review by joa-quim — APPROVED** [review]
